ledger/metrics/src/main/scala/com/daml/telemetry/TelemetryContext.scala

```diff
@@ -6,20 +6,17 @@ package com.daml.telemetry
 import java.util.{HashMap => jHashMap, Map => jMap}
 
 import com.daml.dec.DirectExecutionContext
-import io.grpc.MethodDescriptor
-import io.opentelemetry.api.common.AttributeKey
 import io.opentelemetry.api.trace.Span
 import io.opentelemetry.context.Context
-import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
 
 import scala.concurrent.Future
 import scala.util.{Failure, Success}
 
 trait TelemetryContext {
 
-  /** Sets or replaces the value of attribute `key` to `value`.
+  /** Sets or replaces the value of `attribute` to `value`.
     */
-  def setAttribute[T](key: AttributeKey[T], value: T): TelemetryContext
```

Comment:
@miklos-da are you fine with supporting just the String values here? 

There are 2-3 places in the oem-integration-kit repo where other types have been used, which I easily changed (to `String`s). I'm not sure if, potentially, the only drawback is in the UI that will represent everything as `String`s. 

Adding a type parameter to the `SpanAttribute` (which I already checked) complicates things a bit. I could also leave this particular method as-is. However, it results in an additional dependency and exposing the raw types from the opentelementry lib.

Reply:
Yes, it's OK to simplify it now. We can still go back to supporting typed span attributes if requested by users of `TelemetryContext`.

Reply:
No special concern from me to limit ourselves to `String`.
The generic support is only used for one attribute in the oem-integration-kit repo, and I don't think it is really needed.
There is already a comment in the code stating that putting the values into a `String`, instead of using `List[String]`, might be as good.
And the trace UI is mainly text search based anyway.